pagopa/io-app
# chore: [IOBP-187,IOBP-188,IOBP-189,IOBP-190,IOBP-191] A11y fixes in payment method details

## Short description
added a11y labels for voiceovers in various payment details method screens

## List of changes proposed in this pull request
- small refactors and touchups where needed for the updates
- added labels
- added I18n keys
- removed obsolete I18n key
- added new titles for said screen
- snapshot updates due to component edits

## How to test
with the dev-server, update its config to return all kinds of payment methods, then (IOS) run the app, open Xcode's a11y inspector and check that the voiceover displays the text requested in the linked issues
XCode a11y inspector menu:
<img width="496" alt="Screenshot 2023-08-29 at 15 31 03" src="https://github.com/pagopa/io-app/assets/60693085/793749ed-bffd-42fb-af39-7f33e0b617c2">
